src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DocumentChangedSynchronizationService.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.OmniSharpPlugin;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common
+{
+    [Shared]
+    [Export(typeof(IRazorDocumentChangeListener))]
+    [Export(typeof(IOmniSharpProjectSnapshotManagerChangeTrigger))]
+    internal class DocumentChangedSynchronizationService : IRazorDocumentChangeListener, IOmniSharpProjectSnapshotManagerChangeTrigger
+    {
+        private readonly OmniSharpForegroundDispatcher _foregroundDispatcher;
+        private OmniSharpProjectSnapshotManagerBase _projectManager;
+
+        [ImportingConstructor]
+        public DocumentChangedSynchronizationService(OmniSharpForegroundDispatcher foregroundDispatcher)
+        {
+            if (foregroundDispatcher is null)
```

Comment:
I don't have strong opinions about `is null`s but it'd be nice if we kept the null checks consistent across the repo.

Reply:
Agree however it's not worth the time in fixing all of them given that's how Roslyn generates these now 😢